### ISSUE TITLE
TableMergeError while doing EventTable.read() due to incompatible types

### ISSUE THREAD

**Issue by rutujagurav** [gwpy/gwpy#1198]
I am trying to get triggers from certain channels for O2. And I get an astropy TableMergeError.
Example:
```
trigs = gwtrigfind.find_trigger_files(H1:LSC-MICH_IN1_DQ, 'omicron', gps0, gps1)
glitches = EventTable.read(trigs, tablename='sngl_burst')
```
Traceback (most recent call last):
  File "get-aux-trigs.py", line 71, in <module>
    glitches = EventTable.read(trigs, tablename='sngl_burst')
  File "/usr/lib/python2.7/site-packages/gwpy/table/table.py", line 243, in read
    return io_read_multi(vstack, cls, source, *args, **kwargs)
  File "/usr/lib/python2.7/site-packages/gwpy/io/mp.py", line 112, in read_multi
    return flatten(out)
  File "/usr/lib64/python2.7/site-packages/astropy/table/operations.py", line 253, in vstack
    out = _vstack(tables, join_type, col_name_map)
  File "/usr/lib64/python2.7/site-packages/astropy/table/operations.py", line 789, in _vstack
    out_descrs = get_descrs(arrays, col_name_map)
  File "/usr/lib64/python2.7/site-packages/astropy/table/operations.py", line 545, in get_descrs
    .format(names[0], tme._incompat_types))
astropy.table.np_utils.TableMergeError: The 'ifo' columns have incompatible types: 
['unicode64', ... , 'float64', ... , 'unicode64']

How do I cast all the columns in a consistent type? The documentation for the read method doesn't provide much help in this regard. 

I didn't face this error for a different channel, say, H1:LSC-MCL_IN1_DQ.

**Comment by duncanmmacleod** [comment]
@rutujagurav, can you give specific values for `gps0` and `gps1` to allow us to reproduce the failure?

**Comment by rutujagurav** [comment]
I am getting all triggers for O2 at LHO, so I used gps0 = 1164556817-10 and gps1 = 1187733618+10

**Comment by rutujagurav** [comment]
@duncanmmacleod An update - I usually pull all the rows and columns using EventTable.read() and then discard the ones which I don't need. 
But given this issue, I decided to specify the columns that I want in read() -> ['start_time','start_time_ns','peak_time','peak_time_ns','duration','peak_frequency','central_freq','bandwidth','snr','amplitude']
And it didn't throw this error, so perhaps the issue lies in the remaining columns. 
Unfortunately, the TableMergeError message is not very helpful in determining exactly which column across the trigger files is causing the error when astropy tries to vstack.

**Comment by duncanmmacleod** [comment]
I believe this was fixed in #1204.
